### PR TITLE
Automate labeling of new issues for triage

### PR DESCRIPTION
**Pull request by aeschright**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Adds a GitHub action workflow to automatically apply a `needs-triage` label to new issues for filtering.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
N/A
```

### CODE REVIEW

File: .github/main.workflow [hunk 1]
```diff
@@ -20,3 +20,14 @@ action "pr-label" {
         LABEL_SPEC_FILE = ".github/PULL_REQUEST_LABELS.yml"
     }
 }
+
+workflow "Issue triage" {
+  on = "issue"
```
Comment (bflad) [inline .github/main.workflow]: I believe the (confusing) correct value for this argument is `issues` according to the [GitHub Actions documentation](https://developer.github.com/actions/managing-workflows/workflow-configuration-options/#events-supported-in-workflow-files):

```suggestion
  on = "issues"
```

It appears the documentation in the https://github.com/actions/github repository has the same issue 😅

File: .github/main.workflow [hunk 2]
```diff
@@ -5,7 +5,7 @@ workflow "Pull Request Updates" {
 
 action "pr-filter-sync" {
     uses = "actions/bin/filter@master"
-    args = "action 'opened|review_requested|synchronize'"
```
Comment (bflad) [inline .github/main.workflow]: `review_requested` was adding in here due to GitHub Actions errors previously, but maybe those are fixed now. 🤞

File: .github/main.workflow (hunk 1 above)
Comment (aeschright) [reply .github/main.workflow]: Ha! Thanks for catching that.

File: .github/main.workflow (hunk 2 above)
Comment (aeschright) [reply .github/main.workflow]: 👍

**Pull request merged**